prow/plugins/updateconfig/updateconfig_test.go

```diff
@@ -723,6 +723,280 @@ func TestUpdateConfig(t *testing.T) {
+				Maps: map[string]plugins.ConfigMapSpec{
+					"prow/*.yaml": {
+						Name: "config",
+					},
+				},
+			},
+		},
+		{
+			name:        "converts a binary key to a text key when it becomes text",
+			prAction:    github.PullRequestActionClosed,
+			merged:      true,
+			mergeCommit: "12345",
+			changes: []github.PullRequestChange{
+				{
+					Filename:  "prow/becoming-text.yaml",
+					Status:    "modified",
+					Additions: 1,
+				},
+			},
+			existConfigMaps: []runtime.Object{
+				&coreapi.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "config",
+						Namespace: defaultNamespace,
+					},
+					BinaryData: map[string][]byte{
+						"becoming-text.yaml": []byte("not-yet-text\x00\xFF\xFF"),
+					},
+				},
+			},
+			expectedConfigMaps: []*coreapi.ConfigMap{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "config",
+						Namespace: defaultNamespace,
+					},
+					Data: map[string]string{
+						"becoming-text.yaml": "now-text",
+					},
+					BinaryData: map[string][]uint8{},
+				},
+			},
+			config: &plugins.ConfigUpdater{
+				Maps: map[string]plugins.ConfigMapSpec{
+					"prow/*.yaml": {
+						Name: "config",
+					},
+				},
+			},
+		},
+		{
+			name:        "simultaneously converts text to binary and binary to text",
+			prAction:    github.PullRequestActionClosed,
+			merged:      true,
+			mergeCommit: "12345",
+			changes: []github.PullRequestChange{
+				{
+					Filename:  "prow/becoming-text.yaml",
+					Status:    "modified",
+					Additions: 1,
+				},
+				{
+					Filename:  "prow/becoming-binary.yaml",
+					Status:    "modified",
+					Additions: 1,
+				},
+			},
+			existConfigMaps: []runtime.Object{
+				&coreapi.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "config",
+						Namespace: defaultNamespace,
+					},
+					BinaryData: map[string][]byte{
+						"becoming-text.yaml": []byte("not-yet-text\x00\xFF\xFF"),
+					},
+					Data: map[string]string{
+						"becoming-binary.yaml": "not-yet-binary",
+					},
+				},
+			},
+			expectedConfigMaps: []*coreapi.ConfigMap{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "config",
+						Namespace: defaultNamespace,
+					},
+					BinaryData: map[string][]byte{
+						"becoming-binary.yaml": []byte("now-binary\x00\xFF\xFF"),
+					},
+					Data: map[string]string{
+						"becoming-text.yaml": "now-text",
+					},
+				},
+			},
+			config: &plugins.ConfigUpdater{
+				Maps: map[string]plugins.ConfigMapSpec{
+					"prow/*.yaml": {
+						Name: "config",
+					},
+				},
+			},
+		},
+		{
+			name:        "correctly converts to binary when gzipping",
+			prAction:    github.PullRequestActionClosed,
+			merged:      true,
+			mergeCommit: "12345",
+			changes: []github.PullRequestChange{
+				{
+					Filename:  "prow/config.yaml",
+					Status:    "modified",
+					Additions: 1,
+				},
+				{
+					Filename:  "prow/config.yaml",
+					Status:    "modified",
+					Additions: 1,
+				},
```

Comment:
nit: Duplicate PullRequestChange here and below.

Reply:
Fixed.